path_trie.go

```diff
@@ -64,7 +72,7 @@ func (trie *PathTrie) Put(key string, value interface{}) bool {
 			if node.children == nil {
 				node.children = map[string]*PathTrie{}
 			}
-			child = NewPathTrie()
+			child = NewPathTrie(&PathTrieConfig{trie.segmenter})
```

Comment:
I believe this is unnecessary, but in order to avoid memory allocation as much as possible, should we introduce a new constructor to create trie node without either PathTrieConfig or Segmenter?